### PR TITLE
Migrate `cosh` and `cosh_` from TH to ATen (CUDA)

### PR DESCRIPTION
**Pull request by pdlive215**
The Sep 25 commit addresses Issue [#24546](https://github.com/pytorch/pytorch/issues/24546): migrating `cosh` and `cosh_` from TH to ATen (CUDA)

### CODE REVIEW

**Comment by xuhdev** [comment]
Could you make two separate PRs for easier review? Thanks!

**Comment by pdlive215** [comment]
I already submitted a PR for the Sep 20 commit. What's the simplest way to remove that commit from this PR?

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -717,15 +717,12 @@
...
-    CPU: _cosh__cpu
-    CUDA: _cosh__cuda
 
 - func: cosh.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)
   supports_named_tensor: True
   dispatch:
-    CPU: _cosh_out_cpu
-    CUDA: _cosh_out_cuda
+    CPU: _cosh_out
+    CUDA: _cosh_out
```
Comment (xuhdev) [inline aten/src/ATen/native/native_functions.yaml]: The leading underscores might lead to build errors

**Review by xuhdev — COMMENTED** [review]
Also, please rebase to resolve conflict

File: aten/src/ATen/native/native_functions.yaml (hunk 1 above)
Comment (pdlive215) [reply aten/src/ATen/native/native_functions.yaml]: What would you advise? I believe this is the correct dispatch, and there are many other operators in `native_functions.yaml` that use leading underscores.

Comment (xuhdev) [reply aten/src/ATen/native/native_functions.yaml]: These two lines point to your newly defined function in UnaryOps.cpp, whose name do not start with `_`. That's why I said that leaving the underscores here will likely to fail the build.

File: aten/src/ATen/native/UnaryOps.cpp [hunk 2]
```diff
@@ -65,6 +65,10 @@ Tensor& ceil_out(Tensor& result, const Tensor& self) { return unary_op_impl_out(
 Tensor ceil(const Tensor& self) { return unary_op_impl(self, at::ceil_out); }
 Tensor& ceil_(Tensor& self) { return unary_op_impl_(self, at::ceil_out); }
 
+Tensor& cosh_out(Tensor& result, const Tensor& self) { return unary_op_impl_out(result, self, cosh_stub); }
+Tensor cosh(const Tensor& self) { return cosh_op_impl(self, at::cosh_out); }
```
Comment (xuhdev) [inline aten/src/ATen/native/UnaryOps.cpp]: `unary_op_impl`, not `cosh_op_impl`.

Comment (pdlive215) [reply aten/src/ATen/native/UnaryOps.cpp]: derp

File: aten/src/ATen/native/native_functions.yaml [hunk 3]
```diff
@@ -739,15 +739,12 @@
...
-  dispatch:
-    CPU: _cosh__cpu
-    CUDA: _cosh__cuda
 
 - func: cosh.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)
   supports_named_tensor: True
   dispatch:
-    CPU: _cosh_out_cpu
-    CUDA: _cosh_out_cuda
+    CPU: cosh_out
```
Comment (ifedan) [inline aten/src/ATen/native/native_functions.yaml]: can be removed

Comment (pdlive215) [reply aten/src/ATen/native/native_functions.yaml]: Which lines can be removed?

Comment (ifedan) [reply aten/src/ATen/native/native_functions.yaml]: you can remove:
dispatch:
    CPU: cosh_out
    CUDA: cosh_out

**Comment by xuhdev** [comment]
Are you still working on this? You need to [rebase](https://git-scm.com/docs/git-rebase) -- the codebase that you made changes on are too old

**Comment by ShawnZhong** [comment]
Seems that `cosh` and `cosh_` has been migrated in https://github.com/pytorch/pytorch/pull/36654

**Comment by zou3519** [comment]
Closing this due to @ShawnZhong's comment.

**Pull request closed without merging**